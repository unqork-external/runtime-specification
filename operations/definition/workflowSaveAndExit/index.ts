export * from './workflowSaveAndExit.model'
export * from './workflowSaveAndExit.options'
