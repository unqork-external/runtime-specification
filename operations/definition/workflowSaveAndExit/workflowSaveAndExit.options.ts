import { Description, Ignore, Required } from '@tsed/schema'

import { OperationOptions } from '../../interface'

export class SaveAndExitAlertSetting {
  @Description('message for save and exit modal')
  saveAndExitMessage?: string

  @Description('The cancel button text for save and exit modal.')
  cancelButtonText: string = 'Cancel'

  @Description('The confirmation button text for save and exit modal.')
  confirmButtonText: string = 'OK'
}

export class WorkflowSaveAndExitOperationOptions extends OperationOptions {
  // This can be removed once BaseOperationOperations is refactored.
  @Description('targetKey refers to the intended target to manipulate.')
  // we can ignore properties with type never - Remove this decorator if this type is changed
  @Ignore()
  targetKey?: never

  @Required()
  @Description('current step of the workflow')
  currentStepPath: string

  @Description('save and exit modal alert setting including save and exit message, button texts')
  saveAndExitAlertSetting: SaveAndExitAlertSetting
}
