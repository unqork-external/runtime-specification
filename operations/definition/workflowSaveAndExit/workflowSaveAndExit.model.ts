import { DiscriminatorValue, Required } from '@tsed/schema'

import { WorkflowSaveAndExitOperationOptions } from './workflowSaveAndExit.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.WORKFLOW_SAVE_AND_EXIT)
export class WorkflowSaveAndExitOperation extends Operation<OperationTypes.WORKFLOW_SAVE_AND_EXIT> {
  @Required()
  public options: WorkflowSaveAndExitOperationOptions = new WorkflowSaveAndExitOperationOptions()
}
