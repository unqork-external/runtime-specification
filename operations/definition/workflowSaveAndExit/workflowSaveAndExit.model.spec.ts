import { WorkflowSaveAndExitOperation } from './workflowSaveAndExit.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: WorkflowSaveAndExit Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(WorkflowSaveAndExitOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toStrictEqual({
      type: 'object',
      properties: {
        creatorSummary: {
          description: 'A detailed summary of the operation',
          type: 'string',
        },
        type: {
          type: 'string',
          const: 'WORKFLOW_SAVE_AND_EXIT',
          examples: ['WORKFLOW_SAVE_AND_EXIT'],
        },
        options: {
          $ref: '#/definitions/WorkflowSaveAndExitOperationOptions',
        },
        name: {
          description: 'Name of the operation',
          type: 'string',
        },
      },
      required: ['options'],
      definitions: {
        SaveAndExitAlertSetting: {
          properties: {
            confirmButtonText: {
              description: 'The confirmation button text for save and exit modal.',
              type: 'string',
            },
            cancelButtonText: {
              description: 'The cancel button text for save and exit modal.',
              type: 'string',
            },
            saveAndExitMessage: {
              description: 'message for save and exit modal',
              type: 'string',
            },
          },
          type: 'object',
        },
        WorkflowSaveAndExitOperationOptions: {
          properties: {
            currentStepPath: {
              description: 'current step of the workflow',
              minLength: 1,
              type: 'string',
            },
            saveAndExitAlertSetting: {
              $ref: '#/definitions/SaveAndExitAlertSetting',
            },
          },
          required: ['currentStepPath'],
          type: 'object',
        },
      },
    })
  })
})
