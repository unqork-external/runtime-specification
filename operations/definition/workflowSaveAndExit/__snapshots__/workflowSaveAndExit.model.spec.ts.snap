// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Operation Model: WorkflowSaveAndExit Tests should match this JSON Schema 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "SaveAndExitAlertSetting": {
      "properties": {
        "cancelButtonText": {
          "description": "The cancel button text for save and exit modal.",
          "type": "string",
        },
        "confirmButtonText": {
          "description": "The confirmation button text for save and exit modal.",
          "type": "string",
        },
        "saveAndExitMessage": {
          "description": "message for save and exit modal",
          "type": "string",
        },
      },
      "type": "object",
    },
    "WorkflowSaveAndExitOperationOptions": {
      "properties": {
        "currentStepPath": {
          "description": "current step of the workflow",
          "minLength": 1,
          "type": "string",
        },
        "saveAndExitAlertSetting": {
          "$ref": "#/definitions/SaveAndExitAlertSetting",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
      },
      "required": [
        "currentStepPath",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/WorkflowSaveAndExitOperationOptions",
    },
    "type": {
      "const": "WORKFLOW_SAVE_AND_EXIT",
      "examples": [
        "WORKFLOW_SAVE_AND_EXIT",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
